sdk/dotnet/Pulumi/Deployment/Deployment_Invoke.cs

```diff
@@ -112,6 +112,7 @@ private async Task<SerializationResult> InvokeRawAsync(string token, Serializati
                 Version = options?.Version ?? "",
                 Args = serialized,
                 AcceptResources = !_disableResourceReferences,
+                PluginDownloadURL = options?.PluginDownloadURL ?? "",
```

Comment:
Need to pass the URL set on the `Resource`'s `_pluginDownloadURL` field to `CallRequest` in `Deployment_Call.cs` as well:

https://github.com/pulumi/pulumi/blob/3715afa5907db5d80dab8c107a8c03f0268169c8/sdk/dotnet/Pulumi/Deployment/Deployment_Call.cs#L79